main/src/library/flix/core/RedBlackTree.flix

```diff
@@ -0,0 +1,227 @@
+///
+/// The color of a red-black tree node.
+///
+enum Color {
+    case Red
+    case Black
+    case DoubleBlack
+}
+
+///
+/// The red-black tree type.
+///
```

Comment:
It would be nice if you could list the 5 RB tree invariants here. (Or whichever invariants are relevant for this tree).